extensions/applicationinsights-dependencies-js/src/ajax.ts

```diff
@@ -1186,6 +1186,9 @@ export class AjaxMonitor extends BaseTelemetryPlugin implements IDependenciesPlu
 
                 if (input instanceof Request) {
                     ajaxData.requestUrl = input ? input.url : "";
+                } else if (input === "" ) {
+                    const location = getLocation();
+                    ajaxData.requestUrl = location?.href?.split("#")[0] || input;
```

Comment:
We tend to not use the TypeScript `?.` operator as this causes a lot of JavaScript to get generated and we are always attempting to keep the size of the generated JS code to a minimum.

Suggest maybe something like the following so that we handle the case where input.url === ""
```ts
let requestUrl: string;
if (input instanceof Request) {
    requestUrl = (input||{}).url || "";
} else {
    requestUrl = input;
}

// Handle case where either the input (as a string) or the input.url is an empty string
if (requestUrl === "") {
    const location = getLocation();
    if (location && location.href) {
        requestUrl = strSplit(location.href, "#")[0];
    }
}

ajaxData.requestUrl = requestUrl;
```

Where `strSplit` is imported from `@nevware21/ts-utils`

Reply:
ah, so not a "does it work" thing, but "how much JS gets generated" - got it.

Reply:
oh, that's a good point too re: "what if `input.url === ""`"

Reply:
`new Request("")` appears to be setting `url` to the current URL itself, but it doesn't hurt to handle that here.

Reply:
update pushed

Reply:
> ah, so not a "does it work" thing, but "how much JS gets generated" - got it.

Will need to create a work item to track those down and remove ☹️